api/v1/Store.php

```diff
@@ -404,87 +429,105 @@ private function LoadJSON(string $jsonString, string $filename, string $dataStor
-                            continue;
-                        }
-                        $this->insertMultiPolygon(
-                            $multi,
-                            $layerId,
-                            $encodedFeatureData,
-                            $countryId,
-                            $type,
-                            $mspId,
-                            $filename
-                        );
-                    }
-                } elseif (strcasecmp($geometryData["type"], "Point") == 0) {
+            } elseif (strcasecmp($geometryData["type"], "MultiPoint") == 0) {
+                $this->InsertGeometry(
+                    $layerId,
+                    json_encode($geometryData["coordinates"]),
+                    $encodedFeatureData,
+                    $countryId,
+                    $type,
+                    $mspId,
+                    0,
+                    $filename
+                );
+            } elseif (strcasecmp($geometryData["type"], "MultiLineString") == 0) {
+                foreach ($geometryData["coordinates"] as $line) {
                     $this->InsertGeometry(
                         $layerId,
-                        json_encode(array($geometryData["coordinates"])),
+                        json_encode($line),
                         $encodedFeatureData,
                         $countryId,
                         $type,
                         $mspId,
                         0,
                         $filename
                     );
-                } elseif (strcasecmp($geometryData["type"], "MultiLineString") == 0) {
-                    foreach ($geometryData["coordinates"] as $line) {
-                        $this->InsertGeometry(
-                            $layerId,
-                            json_encode($line),
-                            $encodedFeatureData,
-                            $countryId,
-                            $type,
-                            $mspId,
-                            0,
-                            $filename
-                        );
-                    }
-                } else {
-                    throw new Exception(
-                        "Encountered unknown feature type ".$geometryData["type"]." in layer ".$filename
-                    );
                 }
+            } else {
+                throw new Exception(
+                    "Encountered unknown feature type ".$geometryData["type"]." in layer ".$filename
+                );
             }
         }
     }
 
+    /**
+     * @throws Exception
+     */
+    private function getLayerId(
+        string $layerName,
+        string $layerGeoType,
+        string $layerGroup
+    ): int {
+        $checkExists = Database::GetInstance()->query(
+            "SELECT layer_id FROM layer WHERE layer_name = ?",
+            array($layerName)
+        );
+        if (!empty($checkExists) && isset($checkExists[0]["layer_id"])) {
```

Comment:
if layer_id is not allowed to be 0, you can just do this:

```suggestion
        if (!empty($checkExists[0]["layer_id"])) {
```

Reply:
Yes, totally, did this.